### ISSUE TITLE
Task: Add rules to keep accessibility and performances as high as possible

### ISSUE THREAD

**Issue by legrandgrand** [leaderboardsgg/leaderboard-site#95]
### What are you wanting to be added?

speedrun.website should be opened to everyone  and every computer, every kind of connection. As such, we need to make sure that this stays a priority whenever we create new features or when we change the UI.

In order to do so, we need to make sure to respect as many rules of the WCAG (https://www.w3.org/WAI/standards-guidelines/wcag/ ) which gives all the standard for accessibility, and at least stay at AA standard. To do so, I propose to setup eslint a11y (https://www.npmjs.com/package/eslint-plugin-vue-a11y) on the project, which follows these rules.

We should also use Lighthouse on the CI ( https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/getting-started.md ) to make sure we stay high on both performances and accessibility.

TL;DR:
* Setup a11y rules with Eslint for local development
* Setup Lighthouse on the CI to measure performances and accessibility
* Fix whatever else needs to be fixed to meet our standard: I advice going for AA.

### What might break?

Nothing should break; if anything it should make fix stuff for accessibility purpose :D

### Relevant Issues

N/A
